### PR TITLE
feat(perf-issues): Populate performance issue metadata with type and value

### PR DESCRIPTION
**Pull request by udameli**
Issue details in the issues feed is populated based on the `metadata`. This adds the following metadata: `{ "title": "N+1 Query: <span.description>", "type": "N+1 Query", "value": "<span.description>"}`

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -1172,6 +1172,11 @@ def materialize_metadata(data, event_type, event_metadata):
     }
 
 
+def get_metadata_for_performance_problem(problem: PerformanceProblem):
+    # TODO make type dynamic, pull it from group type
+    return {"type": "N+1 Query", "value": problem.desc, "title": f"N+1 Query: {problem.desc}"}
+
```
Comment (udameli) [inline src/sentry/event_manager.py:1178]: i couldn't just use `metadata` on the `event` because each performance issue would have a different metadata

**Review by evanpurkhiser — DISMISSED** [review]

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -2047,8 +2052,8 @@ def _save_aggregate_performance(jobs: Sequence[Performance_Job], projects):
 
                     is_new = False
 
-                    description = performance_problems_by_fingerprint[existing_grouphash.hash].desc
-                    kwargs["data"]["metadata"]["title"] = f"N+1 Query: {description}"
+                    problem = performance_problems_by_fingerprint[existing_grouphash.hash]
+                    kwargs["data"]["metadata"] = get_metadata_for_performance_problem(problem)
```
Comment (wedamija) [inline src/sentry/event_manager.py:2056]: I don't think we can overwrite all the metadata like this, we still need the base metadata from the event

Comment (evanpurkhiser) [reply src/sentry/event_manager.py:2056]: Will let dan take over the review for this

Comment (udameli) [reply src/sentry/event_manager.py]: yeah I wasn't sure about it either 👍 
I've added a change where performance problem metadata is injected to the existing metadata

File: src/sentry/event_manager.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/event_manager.py:1193]: I just realised that if we create multiple groups for a job that we're actually modifying the same `metadata` dict for each one. I don't think it's causing problems, but just to be safe let's copy `metadata` before we start modifying it.

Comment (udameli) [reply src/sentry/event_manager.py:1193]: 👍 added that

File: src/sentry/event_manager.py [hunk 3]
```diff
@@ -2033,10 +2041,14 @@ def _save_aggregate_performance(jobs: Sequence[Performance_Job], projects):
                         metric_tags["create_group_transaction.outcome"] = "no_group"
 
                         problem = performance_problems_by_fingerprint[new_grouphash]
-                        kwargs["type"] = problem.type.value
-                        kwargs["data"]["metadata"]["title"] = f"N+1 Query: {problem.desc}"
+                        group_kwargs = kwargs
```
Comment (wedamija) [inline src/sentry/event_manager.py:2044]: We still need to copy this first, `kwargs.copy()`

Comment (udameli) [reply src/sentry/event_manager.py]: oh excuse my JS habits 😶‍🌫️

**Review by wedamija — APPROVED** [review]

**Pull request merged**